sdk/program/src/vote/state/mod.rs

```diff
@@ -818,6 +873,60 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_vote_deserialize_into() {
+        // base case
+        let target_vote_state = VoteState::default();
+        let vote_state_buf =
+            bincode::serialize(&VoteStateVersions::new_current(target_vote_state.clone())).unwrap();
+
+        let mut test_vote_state = VoteState::default();
+        VoteState::deserialize_into(&vote_state_buf, &mut test_vote_state).unwrap();
+
+        assert_eq!(target_vote_state, test_vote_state);
+
+        // variant
+        for _ in 0..1000 {
+            let size_hint = VoteStateVersions::size_hint(0).0 * 4;
```

Comment:
`size_hint` returns a tuple with a lower bound and possibly an upper bound (here `None` because the struct can be any size). i just gave `arbitrary` 4x the minimum size to work with so it typically touches all the fields

Reply:
...and it turns out this always returns `(0, None)` anyway so this is actually wrong